asset/src/job_metric_api/api.ts

```diff
@@ -0,0 +1,208 @@
+import {
+    OperationAPI, WorkerContext, ExecutionConfig
+} from '@terascope/job-components';
+import os from 'os';
+import {
+    Gauge, Counter, Histogram, Summary
+} from 'prom-client';
+import { JobMetricAPIConfig, JobMetricsApi } from './interfaces';
+import createExporter from './exporter';
+
+export default class Metrics extends OperationAPI<JobMetricAPIConfig> {
+    metricList: any;
+    default_labels!: Record<string, string>;
+    prefix: string;
+
+    constructor(
+        workerContext: WorkerContext,
+        apiConfig: JobMetricAPIConfig,
+        executionConfig: ExecutionConfig
+
+    ) {
+        super(workerContext, apiConfig, executionConfig);
+        this.metricList = {};
+        this.default_labels = {
+            ex_id: executionConfig.ex_id,
+            job_id: executionConfig.job_id,
+            job_name: executionConfig.name,
+            name: workerContext.sysconfig.teraslice.name,
+        };
+        this.prefix = 'teraslice_job_';
+        this.setPodName();
+    }
+
+    setPodName(): void {
+        // in a pod the hostname is the pod name
+        const host = os.hostname();
+        if (host.startsWith('ts-wkr-')) {
+            this.default_labels.pod_name = host;
+        }
+    }
+
+    /**
+     * [set metric to value]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value]
+     * @return {Promise<void>}
+     */
+    async set(name: string, labelValues: Array<string>, value: number): Promise<void> {
```

Comment:
They also require ordered label strings, which is not great use, and will lead to human error. Ideally they should named, so instead of having to keep the order of `['value1', 'value2', 'value3']`, you can do `{ label2: 'value2', label1: 'value1', label3: 'label3' }` which is more intuitive when using this API.